exasol/slc/internal/tasks/push/docker_push.py

```diff
@@ -17,33 +22,48 @@
 
 
 class DockerFlavorsPush(FlavorsBaseTask, DockerPushParameter):
-    goals = luigi.ListParameter()
+    goals: Tuple[str, ...] = luigi.ListParameter()  # type: ignore
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args, **kwargs) -> None:
         self.image_info_futures = None
         super().__init__(*args, **kwargs)
 
-    def register_required(self):
-        tasks = self.create_tasks_for_flavors_with_common_params(  # type: ignore
+    def register_required(self) -> None:
+        tasks: Dict[
+            str, DockerFlavorPush
+        ] = self.create_tasks_for_flavors_with_common_params(
             DockerFlavorPush
-        )  # type: Dict[str,DockerFlavorPush]
+        )  # type: ignore
         self.image_info_futures = self.register_dependencies(tasks)
 
-    def run_task(self):
-        image_infos = self.get_values_from_futures(self.image_info_futures)
+    def run_task(self) -> None:
+        image_infos: Dict[str, List[ImageInfo]] = self.get_values_from_futures(
+            self.image_info_futures
+        )
+        assert isinstance(image_infos, dict)
+        assert all(isinstance(x, str) for x in image_infos.keys())
+        assert all(isinstance(x, list) for x in image_infos.values())
+        assert all(isinstance(y, ImageInfo) for x in image_infos.values() for y in x)
+
         self.return_object(image_infos)
 
 
 class DockerFlavorPush(DockerFlavorBuildBase, DockerPushParameter):
-    goals = luigi.ListParameter()
+    goals: Tuple[str, ...] = luigi.ListParameter()  # type: ignore
+
+    def __init__(self, *args, **kwargs) -> None:
+        self.image_info_futures = None
+        super().__init__(*args, **kwargs)
 
-    def get_goals(self):
-        return self.goals
+    def get_goals(self) -> Set[str]:
+        return set(self.goals)
 
-    def run_task(self):
+    def run_task(self) -> Generator[BaseTask, None, None]:
```

Comment:
Ideally, we should use the correct Task Type for the yield type here, and not `BaseTask`. However, for that we need to fix https://github.com/exasol/integration-test-docker-environment/issues/445 first.